specs/eventing/control-plane.md

```diff
@@ -0,0 +1,1120 @@
+[Trigger](#trigger-lifecycle) or [Subscription](#subscription-lifecycle) will
+need _read_ access to these resources. On Kubernetes, this is most easily
+achieved using role aggregation; on systems using Kubernetes RBAC, resources
+which wish to participate in Addressable resolution should provide the following
+`ClusterRole`:
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: <component>-addressable-resolver
+  labels:
+    duck.knative.dev/addressable: "true"
+rules:
+  - apiGroups: [<resource apiGroup>]
+    resources: [<resource kind >, <resource kind>/status]
+    verbs: ["get", "list", "watch"]
+```
+
+This configuration advice SHALL NOT indicate a requirement for Kubernetes RBAC
+support.
+
+<!-- TODO: define aggregated API roles
+
+Ref:
+- https://github.com/knative/specs/blob/main/specs/eventing/channel.md#aggregated-channelable-manipulator-clusterrole
+- https://github.com/knative/specs/blob/main/specs/eventing/channel.md#aggregated-addressable-resolver-clusterrole
+-
+-->
+
+## Resource Overview
+
+The Knative Eventing API provides a set of primitives to support both
+point-to-point communication channels (`messaging.knative.dev`) and
+content-based event routing (`eventing.knative.dev`). This specification
+describes API interfaces of Knative Eventing resources as well as the supported
+[event routing](./data-plane.md) logic and configuration settings.
+
+At the moment, the Knative Eventing specification does not contemplate any
+non-Kubernetes-backed implementations, and therefore does not specifically
+define the mapping of kubernetes verbs (read, watch, patch, etc) to developer
+roles. See the [Overview documentation](./overview.md) for general definitions
+of the different API objects.
+
+## Error Signalling
+
+<!-- copied from ../serving/knative-api-specification-1.0.md#error-signalling -->
+
+The Knative API uses the
+[Kubernetes Conditions convention](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#typical-status-properties)
+to communicate errors and problems to the user. Each user-visible resource
+described in Resource Overview MUST have a `conditions` field in `status`, which
+must be a list of `Condition` objects of the following form (note that the
+actual API object types may be named `FooCondition` to allow better code
+generation and disambiguation between similar fields in the same `apiGroup`):
+
+<table>
+  <tr>
+   <td><strong>Field</strong>
+   </td>
+   <td><strong>Type</strong>
+   </td>
+   <td><strong>Description</strong>
+   </td>
+   <td><strong>Default Value</strong>
+   </td>
+  </tr>
+  <tr>
+   <td><code>type</code>
+   </td>
+   <td><code>string</code>
+   </td>
+   <td>The category of the condition, as a short, CamelCase word or phrase.
+<p>
+This is the primary key of the Conditions list when viewed as a map.
+   </td>
+   <td>REQUIRED – No default
```

Comment:
let's add a column with REQUIRED vs OPTIONAL for each row so it's clear which need to be there

Reply:
I think this section assumes go/ proto semantics where empty string can be omitted.

Since this is a straight-up copy from `serving/knative-api-specification-1.0.md#error-signalling`, I see three options:

1. Update eventing to clarify, leave serving as-is
2. Update serving as well as eventing, making this PR larger
3. Extract "error conditions" to a `common` directory, and link both serving and eventing to that.

I prefer option 3. Would you like that to be in this PR (along with the updates), or should I defer that to a separate PR?

Reply:
I added some clarification on line 11 of `error-signalling.md`:

> Fields in the
> condition which are not marked as REQUIRED may be omitted to indicate the
> default value (i.e. a Condition which does not include a `status` field is
> equivalent to a `status` of `"Unknown"`)